src/openforms/payments/views.py

```diff
@@ -0,0 +1,306 @@
+        "\n* the `next` parameter must be present"
+        "\n* the `next` parameter must match the CORS policy"
+    ),
+    parameters=[
+        OpenApiParameter(
+            name="uuid",
+            location=OpenApiParameter.PATH,
+            type=OpenApiTypes.STR,
+            description=_("UUID identifying the submission."),
+        ),
+        OpenApiParameter(
+            name="plugin_id",
+            location=OpenApiParameter.PATH,
+            type=OpenApiTypes.STR,
+            description=_(
+                "Identifier of the payment plugin. Note that this is validated "
+                "against the configured available plugins for this particular form."
+            ),
+        ),
+        OpenApiParameter(
+            name="next",
+            location=OpenApiParameter.QUERY,
+            type=OpenApiTypes.URI,
+            description=_(
+                "URL of the form to redirect back to. This URL is validated "
+                "against the CORS configuration."
+            ),
+            required=True,
+        ),
+    ],
+    responses={
+        (200, "application/json"): OpenApiResponse(
+            response=PaymentInfoSerializer,
+            description=_(
+                "Depending on the 'type' send a GET or POST request with the 'data' as Form Data to given 'url'."
+            ),
+        ),
+        (400, "text/html"): OpenApiResponse(
+            response=str, description=_("Bad request. Invalid parameters were passed.")
+        ),
+        (404, "text/html"): OpenApiResponse(
+            response=str,
+            description=_("Not found. The slug did not point to a live submission."),
+        ),
+    },
+)
+class PaymentStartView(PaymentFlowBaseView, GenericAPIView):
+    lookup_field = "uuid"
+    lookup_url_kwarg = "uuid"
+    queryset = Submission.objects.all()
+    serializer_class = PaymentInfoSerializer
+
+    def post(self, request, uuid: str, plugin_id: str):
+        submission = self.get_object()
+        try:
+            plugin = self.register[plugin_id]
+        except KeyError:
+            return HttpResponseBadRequest("unknown plugin")
+
+        if not submission.form.payment_required:
+            return HttpResponseBadRequest("payment not required")
+
+        if plugin_id != submission.form.payment_backend:
+            return HttpResponseBadRequest("plugin not allowed")
+
+        form_url = request.GET.get("next")
+        if not form_url:
+            return HttpResponseBadRequest("missing 'next' parameter")
+
+        if not allow_redirect_url(form_url):
+            logger.warning(
+                "blocked payment start with non-allowed redirect to '%(form_url)s'",
+                {"form_url": form_url},
+            )
+            return HttpResponseBadRequest("redirect not allowed")
+
+        payment = SubmissionPayment.objects.create_for(
+            submission,
+            plugin_id,
+            # TODO pass amount from form
+            Decimal(10),
```

Comment:
As a minimal implementation, allow people to specify the price as a decimal field on the form itself.